### ISSUE TITLE
Checked by default

### ISSUE THREAD

**Issue by eeynard** [RSSchermer/ember-multiselect-checkboxes#33]
Hi guys,

I was wondering if there is an option to make every checkboxes checked by default ? If not, could you help me to find a way to do the job ?

Thanks !

**Comment by RSSchermer** [comment]
Hi @eeynard,

Whether or not a checkbox is checked is determined by whether or not its associated value is contained in the `selection` array. If you want every checkbox to be checked initially, then you could for example bind a copy of the `options` array to the `selection` attribute in the controller's initializer:

```js
// controller.js
import Ember from 'ember';

export default Ember.Controller.extend({
  options: ['a', 'b', 'c'],

  init() {
    // Set the selection property to be a copy of the options array
    this.set('selection', this.get('options').slice());
  }
});
```

```hbs
<!-- template.hbs -->
{{multiselect-checkboxes options=options selection=selection}}
```

Does that help? If not, please explain your use case in a bit more detail and perhaps I can give a more specific answer.

**Comment by eeynard** [comment]
Yes I saw that by digging in your code. But I was struggling with the fact that I am using a filtered PromiseArray as options, something like this : 
```javascript
let filterPromise = Ember.RSVP.filter(objects, object => {
       ...  
});             
    
return DS.PromiseArray.create({
    promise: filterPromise
});
```
My problem was that I wanted to initialize the selection array with the filtered objects fetched from the Promise. But after a few headaches, I just realized that I could use the unfiltered array of objects since it does not matter if there are extra objects in the selection array.

I had a pretty bad time explaining myself here so excuse me if you don't understand everything but I am no longer blocked so I close the issue !

Thank you for your fast answer :smiley:
